Apps/WebClient/src/Server/Services/UserEmailService.cs

```diff
@@ -58,38 +58,44 @@ public class UserEmailService : IUserEmailService
         }
 
         /// <inheritdoc />
-        public bool ValidateEmail(string hdid, Guid inviteKey, string bearerToken)
+        public PrimitiveRequestResult<bool> ValidateEmail(string hdid, Guid inviteKey, string bearerToken)
         {
             this.logger.LogTrace($"Validating email... {inviteKey}");
-            bool retVal = false;
+            PrimitiveRequestResult<bool> retVal = new PrimitiveRequestResult<bool>();
             MessagingVerification? emailInvite = this.messageVerificationDelegate.GetByInviteKey(inviteKey);
 
-            if (emailInvite != null &&
-                emailInvite.HdId == hdid &&
-                !emailInvite.Validated &&
-                !emailInvite.Deleted &&
-                emailInvite.VerificationAttempts < MaxVerificationAttempts &&
-                emailInvite.ExpireDate >= DateTime.UtcNow)
+            if (emailInvite?.HdId != hdid ||
```

Comment:
will this fail if emailInvite is null?

Reply:
null != hdid will enter the if which is the same as the null check "if (null == null || null != hdid)"

Reply:
Yeah I agree, however I would add a emailInvite == null and then the ? is not needed/